qiskit_nature/second_q/properties/dipole_moment.py

```diff
@@ -267,30 +142,21 @@ def interpret(self, result: "EigenstateResult") -> None:
             aux_operator_eigenvalues = result.aux_operator_eigenvalues
 
         for aux_op_eigenvalues in aux_operator_eigenvalues:
-            if aux_op_eigenvalues is None:
-                continue
-
-            if isinstance(aux_op_eigenvalues, list) and len(aux_op_eigenvalues) < 6:
+            if not isinstance(aux_op_eigenvalues, dict):
                 continue
 
-            axes_order = {"x": 0, "y": 1, "z": 2}
-            dipole_moment = [None] * 3
-            for prop in self._dipole_axes.values():
-                moment = aux_op_eigenvalues.get(prop.name, None)
+            dipole_moment = [float("NaN")] * 3
```

Comment:
The dipole tuple type above was `Optional[Number]` - we went from None to NaN (not a Number) though out of curosity I tried `isinstance(float("NaN'), Number)` and despite not being a number it returns True!

Reply:
So are you saying it should actually be type `Number`, dropping the `Optional` since the values will always be numbers now (although possible `NaN`-valued)?